sympy/core/power.py

```diff
@@ -590,6 +590,10 @@ def _eval_is_integer(self):
         if b.is_Number and e.is_Number:
             check = self.func(*self.args)
             return check.is_Integer
+        if e.is_negative and b.is_positive and (b - 1).is_positive:
+            return False
+        if e.is_negative and b.is_negative and (b + 1).is_negative:
+            return False
```

Comment:
Just a side note here:

There are a few places in `Pow._eval_is_*` methods that evaluate e.g. `(b - 1)` and they can be problematic in some situations. We really need #18778